assignments/simple-draw/code.js

```diff
@@ -1,46 +1,35 @@
-/*
- * This code is running in an environment with five functions defined:
- *
- *  drawLine(x1, y1, x2, y2, color, lineWidth)
- *
- *    Draws a line from x1,y1 to x2,y2 using the give color. The fifth argument,
- *    lineWidth, is optional and defaults to 1.
- *
- *  drawCircle(x, y, r, color, lineWidth=1)
- *
- *    Draws a circle centered at x,y with radius r using the given color. The
- *    fith argument, lineWidth, is optional and defaults to 1.
- *
- *  drawRect(x, y, width, height, color, lineWidth=1)
- *
- *    Draws a rectangle starting at x,y with the given width, height, and color.
- *    Positive widths go to the right and negative to the left; positive heights
- *    go down and negative heights go up. The sixth argument, lineWidth, is
- *    optional and defaults to 1.
- *
- *  drawFilledRect(x, y, width, height, color)
- *
- *    Draws a filled rectangle starting at x,y with the given width, height, and
- *    color. Positive widths go to the right and negative to the left; positive
- *    heights go down and negative heights go up.
- *
- *  drawFilledCircle(x, y, r, color)
- *
- *    Draws a filled circle centered at x,y with radius r using the given color.
- *
- * There are also a couple variables that might be useful:
- *
- *  width - the width of the drawing area.
- *  height - the height of the drawing area.
- *
- */
+drawFilledRect (0, 0, 500,600, 'rgba(115,215,255,1)')
+drawFilledRect (0,(2*height/3)+(height/6), width, (2*height/3), 'rgba(0, 64, 0, 1')
+var rect = 1000
+while (rect>= 2*height/3) {
+  drawLine (0, rect, width, rect, 'rgba(0,128,0, 1', 0)
+  var rect = rect - 2
+  while (rect >= 3*height/4){
+  drawLine (0, rect, width, rect, 'rgba(0,128,0, 1', 0)
+   var rect = rect - 1
+  }
+}
 
-drawLine(0, 0, width, height, 'red')
+var rays = 500
+while (rays >= width-2*width/7) {
+  drawLine (0, 0, rays, height, 'rgba(242,242,37,.1', 1)
+  var rays = rays - 1
```

Comment:
Okay. I maybe see now why you felt like `let` doesn't work. Because if you replace all these `var`s with `let` this code indeed won't work. But the problem here is that you have too many `var`s. The `var` on this line is totally redundant. You only need `var` (or `let` or `const`) when you are *declaring* a variable. I.e. causing it to exist. When you change a variable you just change it. I knew `var` was messed up but I didn't know it was so messed up that you could write code like this. This line should be changed like this:

```suggestion
  rays = rays - 1
```

And then if you want to be extra concise you can change it to, either

```javascript
rays--
```

Which is the decrementing version of `x++` that we talked about at the board today. More generally you could write:

```javascript
rays -= 1
```

which lets you specify the amount you want to decrement by. In this case it's one but elsewhere in your code you want to decrement by something else like 6 which would be `rays -= 6`.